### PR TITLE
Add a subscribers file

### PR DESCRIPTION
**Pull request by 2uasimojo**
Adds `subscribers.yaml`, a simple YAML file to keep track of the repositories that are or will be consuming boilerplate conventions.

Adds a `subscriber` utility to do helpful things with these subscribers:
- Report how far behind boilerplate master onboarded subscribers are.
- Look for open boilerplate-related PRs.
- Analyze the consumer's prow configuration.
- Propose boilerplate update PRs.

### CODE REVIEW

**Comment by 2uasimojo** [comment]
Why? I got tired of having to go to the [Jira card](https://issues.redhat.com/browse/SDE-669), and follow the [link](https://github.com/openshift/ops-sop/blob/master/v4/development.asciidoc), and decipher that text, and then remember the ones that aren't on that list yet. This will be slightly painful to maintain by hand in the short term, but will eventually feed into the automated process we've been discussing as the long-term goal.

**Comment by dustman9000** [comment]
How do we _really_ know who subscribes to boilerplate though? I'd much rather prefer a more dynamic way of determining the clones (or possibly forks?) in a similar way that Github determines [unique clones](https://github.com/openshift/boilerplate/graphs/traffic).

**Comment by 2uasimojo** [comment]
That would be cool.

But at the moment there's no way to tell which repositories we're targetting, that haven't done anything yet. That's all we need to accomplish via this file for right now. (I originally had it as a flat text file listing the repositories, but thought it would be more useful this way.)

**Comment by 2uasimojo** [comment]
/hold cancel

This has enough meat to be worth merging.

**Comment by 2uasimojo** [comment]
This is ready to go. The following PRs were all generated by `subscriber propose update`:
- https://github.com/openshift/configure-alertmanager-operator/pull/124
- https://github.com/openshift/custom-domains-operator/pull/17
- https://github.com/openshift/deadmanssnitch-operator/pull/89
- https://github.com/openshift/must-gather-operator/pull/29
- https://github.com/openshift/osd-metrics-exporter/pull/33

That tool correctly accounts for the following edge conditions:
- You don't have a fork
- Your fork is behind upstream
- The consumer is already at the latest boilerplate commit
- The consumer already has an open PR for this update

The other pieces of this PR I've been using regularly to keep my [boilerplate master sheet](https://docs.google.com/spreadsheets/d/1IexmzKQ0h9v0r2S3KZpiKD2bkw_JcCMf4Ey269ky3iM/edit#gid=0) up to date.

File: boilerplate/_lib/subscriber.sh [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+# Prints the commit hash of the specified repository's boilerplate
+# level, or the empty string if the repository is not onboarded.
+#
+# ORG/PROJ: github organization and project name, e.g.
+#           "openshift/my-wizbang-operator".
+last_bp_commit() {
+    local repo=$1
+    local lbc
+    for default_branch in master main; do
+        for lbc_file in last-boilerplate-commit last_boilerplate_commit; do
```
Comment (jharrington22) [inline boilerplate/_lib/subscriber.sh:92]: Was there a change to the file name?

File: boilerplate/_lib/subscriber.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+#
+# Prints how many merge commits behind boilerplate master HASH is. If
+# HASH is empty/unspecified, prints the total number of merge commits in
+# the boilerplate repo.
+commits_behind_bp_master() {
+    local hash=$1
+    local range=master
+    if [[ -n "$hash" ]]; then
+        range=$hash..master
+    fi
```
Comment (jharrington22) [inline boilerplate/_lib/subscriber.sh:112]: This is going to need to support `main` branches

**Review by jharrington22 — COMMENTED** [review]
Couple of small comments but very cool.

File: boilerplate/_lib/subscriber.sh (hunk 1 above)
Comment (2uasimojo) [reply boilerplate/_lib/subscriber.sh:92]: When I wrote this, there were still one or two repositories that hadn't upgraded past #29. At this point we can probably simplify.

File: boilerplate/_lib/subscriber.sh (hunk 2 above)
Comment (2uasimojo) [reply boilerplate/_lib/subscriber.sh:112]: This function is only inspecting the boilerplate repo itself. So unless we decide to migrate boilerplate's default branch to `main` (over my cooling corpse), this is fine as is.

File: boilerplate/_lib/subscriber.sh (hunk 1 above)
Comment (2uasimojo) [reply boilerplate/_lib/subscriber.sh]: Done.

File: boilerplate/_lib/subscriber.sh (hunk 2 above)
Comment (2uasimojo) [reply boilerplate/_lib/subscriber.sh:110]: (I guess technically there's nothing about this function that's bp-specific. But nothing in the subscriber suite needs to know this for a repo other than boilerplate. If that ever changes, we'll deal with it (and rename the function).)

Comment (jharrington22) [reply boilerplate/_lib/subscriber.sh:110]: :+1:

**Review by jharrington22 — APPROVED** [review]
/lgtm

**Pull request merged**